perfkitbenchmarker/configs/benchmark_config_spec.py

```diff
@@ -163,6 +164,71 @@ def __init__(self, **kwargs):
         default=list, item_decoder=_StaticVmDecoder(), **kwargs)
 
 
+class _SparkServiceSpec(spec.BaseSpec):
+  """Configurable options of a VM group.
+
+  Attributes:
+    spark_service_type: string.  pkb_managed or managed_service
+    num_workers: number of workers.
+    machine_type: machine type to use.
+    cloud: cloud to use.
+    project: project to use.
+  """
+
+  def __init__(self, component_full_name, flag_values=None, **kwargs):
+    super(_SparkServiceSpec, self).__init__(component_full_name,
+                                            flag_values=flag_values,
+                                            **kwargs)
+
+  @classmethod
+  def _GetOptionDecoderConstructions(cls):
+    """Gets decoder classes and constructor args for each configurable option.
+
+    Returns:
+      dict. Maps option name string to a (ConfigOptionDecoder class, dict) pair.
+      The pair specifies a decoder class and its __init__() keyword arguments
+      to construct in order to decode the named option.
+    """
+    result = super(_SparkServiceSpec, cls)._GetOptionDecoderConstructions()
+    result.update({
+        'spark_service_type': (option_decoders.EnumDecoder, {
+            'default': spark_service.PROVIDER_MANAGED,
+            'valid_values': [spark_service.PROVIDER_MANAGED,
+                             spark_service.PKB_MANAGED]}),
+        'num_workers': (option_decoders.IntDecoder, {
+            'default': 3, 'min': 1}),
+        'cloud': (option_decoders.EnumDecoder, {
+            'valid_values': providers.VALID_CLOUDS}),
+        'project': (option_decoders.StringDecoder, {'default': None,
+                                                    'none_ok': True}),
+        'machine_type': (option_decoders.StringDecoder, {'default': None,
+                                                         'none_ok': True})})
```

Comment:
Should you take VM Specs (and possibly Disk Specs)?

Most fields wouldn't always be usable, but it seems PKB Managed could use all of them.